File: python Parking lot/parking_lot.py



def is_occupied(parking_space: list[int], index: int)-> bool:
    if parking_space[index] == 1:
        return True
    return False
 
def book_slot(parking_space: list[int], index: int)-> str:
    if not is_occupied(parking_space, index):
        parking_space[index] = 1
        return f" slot{index + 1} Booked successfully".upper()
    return "already Occupied".upper() 

def unoccupy_slot(parking_space: list[int],index: int)-> str:
    if is_occupied(parking_space, index):
        parking_space[index] = 0
        return f"SLOT{index + 1} UNOCCUPIED SUCCESSFULLy" 
    return f"SLOT{index +1} NEVER OCCUPIED"

def occupied_space(parking_space: list[int]) -> list[int]:
   return [index for index, park_space in enumerate(parking_space, 1) if park_space == 1]

def unoccupied_space(parking_space: list[int]) -> list[int]:
   return [index for index, park_space in enumerate(parking_space, 1) if park_space == 0]

File: python Parking lot/ParkingLot.java
import java.util.ArrayList;
import java.util.List;

public class ParkingLot {

    public boolean isOccupied(List<Integer> parkingSpace, int index) {
        if (parkingSpace.get(index) == 1) {
            return true;
        }
        return false;
    }

    public String bookSlot(List<Integer> parkingSpace, int index) {
        if (!isOccupied(parkingSpace, index)) {
            parkingSpace.set(index, 1);
            return String.format(" slot%d Booked successfully", index + 1).toUpperCase();
        }
        return "already Occupied".toUpperCase();
    }

    public String unoccupySlot(List<Integer> parkingSpace, int index) {
        if (isOccupied(parkingSpace, index)) {
            parkingSpace.set(index, 0);
            return String.format("SLOT%d UNOCCUPIED SUCCESSFULLy", index + 1);
        }
        return String.format("SLOT%d NEVER OCCUPIED", index + 1);
    }

    public List<Integer> occupiedSpace(List<Integer> parkingSpace) {
        List<Integer> result = new ArrayList<>();
        for (int i = 0; i < parkingSpace.size(); i++) {
            if (parkingSpace.get(i) == 1) {
                result.add(i + 1);
            }
        }
        return result;
    }

    public List<Integer> unoccupiedSpace(List<Integer> parkingSpace) {
        List<Integer> result = new ArrayList<>();
        for (int i = 0; i < parkingSpace.size(); i++) {
            if (parkingSpace.get(i) == 0) {
                result.add(i + 1);
            }
        }
        return result;
    }
}

